File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/cn/porkchop/pinyougou/sellergoods/service/AreasService.java
package cn.porkchop.pinyougou.sellergoods.service;

import cn.porkchop.pinyougou.pojo.PageResult;
import cn.porkchop.pinyougou.pojo.TbAreas;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface AreasService {

	/**
	 * 返回全部列表
	 * @return
	 */
    List<TbAreas> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
    PageResult findByPagination(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
    void add(TbAreas areas);
	
	
	/**
	 * 修改
	 */
    void update(TbAreas areas);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
    TbAreas findById(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
    void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
    PageResult findWithConditionsByPagination(TbAreas areas, int pageNum, int pageSize);
	
}

File: pinyougou-parent/pinyougou-shop-web/src/main/webapp/js/service/areasService.js
//服务层
app.service('areasService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../areas/findAll.php');
	};
	//分页 
	this.findByPagination=function(page,rows){
		return $http.get('../areas/findByPagination.php?page='+page+'&rows='+rows);
	};
	//查询实体
	this.findById=function(id){
		return $http.get('../areas/findById.php?id='+id);
	};
	//增加 
	this.add=function(entity){
		return  $http.post('../areas/add.php',entity );
	};
	//修改 
	this.update=function(entity){
		return  $http.post('../areas/update.php',entity );
	};
	//删除
	this.dele=function(ids){
		return $http.get('../areas/delete.php?ids='+ids);
	};
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../areas/findWithConditionsByPagination.php?page='+page+"&rows="+rows, searchEntity);
	}    	
});
